### PR TITLE
Token changes

### PR DESCRIPTION
**Pull request by aydun**
Overview
----------------------------------------
A variety of changes affecting the TokenProcessor - see git commits.  These are extracted from PR #12012 and new functionality is used in that PR

There are no user changes here.

### CODE REVIEW

**Comment by aydun** [comment]
@eileenmcnaughton The things left here now are fairly small.  Can we consider these in one PR or do I need to split out further?

File: Civi/Token/TokenRow.php [hunk 1]
```diff
@@ -136,13 +136,13 @@ public function tokens($a = NULL, $b = NULL, $c = NULL) {
    */
   public function customToken($entity, $customFieldID, $entityID) {
     $customFieldName = "custom_" . $customFieldID;
-    $fieldValue = civicrm_api3($entity, 'getvalue', array(
-      'return' => $customFieldName,
-      'id' => $entityID,
+    $record = civicrm_api3($entity, "getSingle", array(
```
Comment (eileenmcnaughton) [inline Civi/Token/TokenRow.php]: not having a return does impact the speed of the api - esp on contribution.get -

File: Civi/Token/AbstractTokenSubscriber.php [hunk 2]
```diff
@@ -71,14 +71,22 @@ public static function getSubscribedEvents() {
```
Comment (eileenmcnaughton) [inline Civi/Token/AbstractTokenSubscriber.php]: So I feel like I understand the changes to this file really well now :-) If you push them up separately I can merge them quick

Comment (aydun) [reply Civi/Token/AbstractTokenSubscriber.php]: Ok, now in #13278

**Comment by eileenmcnaughton** [comment]
@aydun can you please rebase this?

**Comment by aydun** [comment]
@eileenmcnaughton The only piece of this now remaining is #13283 so I'll close this one.  Getting that merged is the last item before rebasing #12012

**Pull request closed without merging**